### ISSUE TITLE
[Bug] "no valid backup files were found" on BackBlaze storage

### ISSUE THREAD

**Issue by Danielss89** [mariadb-operator/mariadb-operator#515]
**Documentation**
- [x] I acknowledge that I have read the relevant [documentation](https://github.com/mariadb-operator/mariadb-operator/tree/main/docs).

**Describe the bug**
I am trying to set up a new mariadb instance from a backup, like the example from the documentation. 
```
apiVersion: mariadb.mmontes.io/v1alpha1
kind: MariaDB
metadata:
  name: uk-merge-test
spec:
  username: test
  database: test

  image: mariadb:11.2
  imagePullPolicy: IfNotPresent

  volumeClaimTemplate:
    storageClassName: db-europe-longhorn-sc
    resources:
      requests:
        storage: 20Gi
    accessModes:
      - ReadWriteOnce

  service:
    type: ClusterIP

  nodeSelector:
    region: europe
    type: db

  bootstrapFrom:
    s3:
      bucket: xxx
      endpoint: s3.eu-central-003.backblazeb2.com
      region: eu-west-1
      prefix: yyy/
      accessKeyIdSecretKeyRef:
        name: mariadb-backblaze
        key: AWS_ACCESS_KEY_ID
      secretAccessKeySecretKeyRef:
        name: mariadb-backblaze
        key: AWS_SECRET_ACCESS_KEY
      tls:
        enabled: true
        caSecretKeyRef:
          name: backblaze-ca
          key: ca
    targetRecoveryTime: 2024-03-26T03:30:25Z
```
As seen in screenshot, backup files does exist, with exact timestamp even.
The S3 part from config is copied directly from the backup yaml file, which works, so that should also work fine.

**Expected behaviour**
Instance would start up and backup would be imported.

**Debug information**
- Related object events:
```bash
mariadb-operator {"level":"info","ts":1711455940.049458,"msg":"starting restore"}    
mariadb-operator {"level":"info","ts":1711455940.0496657,"msg":"configuring S3 backup storage"}     
mariadb-operator {"level":"info","ts":1711455940.0636103,"msg":"obtained target time","time":"2024-03-26 03:30:25 +0000 UTC"} 
mariadb-operator {"level":"error","ts":1711455940.2660177,"msg":"error reading getting target backup","error":"no valid backup files were found","stacktrace":"github.com/mariadb-operator/mariadb-operator/cmd/backup.glob..func2\n\t/app/cmd/backup/restore.go:54\ngithub.com/spf13/cobra.(*Command).execute\n\t/go/pkg/mod/github.com /spf13/cobra@v1.7.0/command.go:944\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\t/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068\ngithub.com/spf13/cobra. (*Command).Execute\n\t/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992\nmain.main\n\t/app/cmd/controller/main.go:361\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:267"}  
```

**Comment by ClementFullOps** [comment]
Just to add some more informations, I encountered a similar issue when trying to restore a backup from s3 in my database.
After digging through the code, it seems that the restore operations does not take the "prefix" into account when looking for viable backup files.
Mainly it checks that the file is named a certain way with a recongnizable pattern, and on some s3 setups the prefix is part of the name. This makes the pattern matching fail, hence no backup is considered "valid" in the end.

Culprit seems to be here: https://github.com/mariadb-operator/mariadb-operator/blob/5be8360b730bdb55a7677fc7e14e41cef79162c9/pkg/backup/backup.go#L69
Where it checks if the string starts with "backup", but does not take into account that the filename may begin by a defined prefix.

@Danielss89 I solved the issue by removing the "prefix" attribute in by backup configuration for now. Or you could copy the requested file at the root of your bucket, which is like removing the prefix.

**Comment by fvillain** [comment]
I've proposed a fix for that, see: #524

**Comment by mmontes11** [comment]
Hey there everybody! Thanks for reporting, the investigation and the PR! 🙏🏻 

Indeed, we have an issue with the prefixes, we will submit a patch for this in `v0.0.28`. For the time being, please refrain from using the `S3` `prefix`.
